crates/net/network/src/peers/manager.rs

```diff
@@ -424,9 +424,12 @@ impl PeersManager {
 
     /// Called for a newly discovered peer.
     ///
-    /// If the peer already exists, then the address and kind will be updated.
-    pub(crate) fn add_peer(&mut self, peer_id: PeerId, addr: SocketAddr) {
-        self.add_peer_kind(peer_id, PeerKind::Basic, addr)
+    /// If the peer already exists, then the address, kind and fork_id will be updated.
+    pub(crate) fn add_peer(&mut self, peer_id: PeerId, addr: SocketAddr, fork_id: Option<ForkId>) {
+        self.add_peer_kind(peer_id, PeerKind::Basic, addr);
+        if let Some(fork_id) = fork_id {
+            self.set_discovered_fork_id(peer_id, fork_id);
+        }
```

Comment:
these are two lookups now, one forst insert Peer then set forkId,

I think we can unify this by extending `add_peer_kind` with `Option<ForkId>` and initialize a `Peer` with that.